### ISSUE TITLE
Wine Compatability (Linux Support)

### ISSUE THREAD

**Issue by maxkleiner** [TurboPack/SynEdit#247]
I am getting an EOSError code by starting an EXE with TSynedit component in it with Wine7: 
**EOSerror SystemError code -2003283966** How can I find the reason why or the meaning of the code?
I tried the dbg debugger and fixed access violations in RAD Studio 11.3 Alexandria but TSynEdit Version from getit in Delphi 10.4 works, the cause is in my opinion is RAD Studio 11.3 Alexandria with TSynEdit (could be an access of Dwrite since the newest SynEdit uses).
Ddb debugger mentioned an unhandled page fault access at the initialisation or start:

![Error_resX_withsynedit_VirtualBox_Kalilinux64_29_11_2023_18_28_25](https://github.com/TurboPack/SynEdit/assets/3393121/891c2b89-38aa-49f9-8370-924bad04b850)
So I decided to check (Resxplor resxplor.exe compile with and without TSynedit component) another small app with TSynedit on it and we got the same error code and also by #PyScripter (which uses Synedit too) we got a hint that it depends on a page fault:

![Error_pyscripter_log_VirtualBox_Kalilinux64_29_11_2023_18_33_21](https://github.com/TurboPack/SynEdit/assets/3393121/428d4473-1854-4a80-8dbc-6fc5e109060b)
See also
https://github.com/maxkleiner/maXbox4/issues/39

**Comment by pyscripter** [comment]
Are you using the latest version of PyScripter?

**Comment by pyscripter** [comment]
Does PyScripter produce an error message with a call stack?

**Comment by maxkleiner** [comment]
Yes the Wine Debugger calls a stack trace window (see above pic 2) and in the meantime I made more testst with synedit based on D 10.4 and synedit D 11.3 , all 64.bit apps before 11.3 (synedit without directwrite) run on Wine even on older versions on wine (1.6 - 8.1) so bug is more or less an exception from SynDwrite or SynEditMiscClasses.?

**Comment by pyscripter** [comment]
> Yes the Wine Debugger calls a stack trace window (see above pic 2)

I mean an error message from PyScripter.

**Comment by maxkleiner** [comment]
No it stops there with the program error details, so I've tried using the cmd line to launch the app as well as just double clicking on the icon wine creates as if it was a windows environment. When I try to use it like a windows environment and just double click, then the same unhandled exception as program error appears.

**Comment by maxkleiner** [comment]
Has to do with issue https://github.com/SynEdit/SynEdit/issues/91
I try next a syn_dwrite disabling:
// DirectWrite
{$IFDEF SYN_DELPHI_XE2_UP}
  {.$DEFINE SYN_DirectWrite}
{$ENDIF}
but I don't know the drawbacks

**Comment by pyscripter** [comment]
Which fork of SynEdit are you using?  You cannot disable DirectWrite with this fork.   Could you please try the Wine workaround suggested in https://github.com/SynEdit/SynEdit/issues/91?

**Comment by maxkleiner** [comment]
The workaround doesnt work, so the same runtime error we find in windows7 is now with wine9 latest version 
![wom7systemerrormx5](https://github.com/TurboPack/SynEdit/assets/3393121/362ff4e3-7c17-4876-bd87-c24042eea853)

**Comment by pyscripter** [comment]
I meant have you tried [this fix](https://github.com/HeidiSQL/HeidiSQL/issues/83#issuecomment-375969129).

**Comment by maxkleiner** [comment]
> I meant have you tried [this fix](https://github.com/HeidiSQL/HeidiSQL/issues/83#issuecomment-375969129).
Yes but one of the workarounds work for me.

**Comment by maxkleiner** [comment]
Does anybody know the error code: -2003283966 (wine9) or -2003292288 (win7) ?

**Comment by maxkleiner** [comment]
We should have at least try to understand which DLL function causes the page fault, tried another synedit, Delphi 12.2 fix with Wine 9, but no success:

![Image](https://github.com/user-attachments/assets/d0ac69cd-2c31-4121-a8cd-3137e8832369)
